### PR TITLE
OCPRHV-635: add tests for ovirt provider

### PR DESCRIPTION
**Pull request by bkhizgiy**
# Assisted Pull Request

Adding subsystem test for oVirt provider, test different installation flows for the new provider option.

## List all the issues related to this PR

- [ ] New Feature
- [ ] Bug fix
- [x] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees
@avishayt 
@yevgeny-shnaidman  
@ori-amizur 
/cc @Gal-Zaidman 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

### CODE REVIEW

File: subsystem/cluster_test.go [hunk 1]
```diff
@@ -4010,3 +4010,346 @@ var _ = Describe("disk encryption", func() {
...
+		}
+
+		for _, host := range c.Hosts {
+			updateProgress(*host.ID, clusterID, models.HostStageDone)
+		}
+		waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout, clusterFinalizingStateInfo)
+		completeInstallationAndVerify(ctx, agentBMClient, clusterID, true)
+	})
+
+	It("host installation progress with ovirt invntory", func() {
```
Comment (avishayt) [inline subsystem/cluster_test.go:4159]:
```suggestion
	It("host installation progress with ovirt inventory", func() {
```

**Review by avishayt — COMMENTED** [review]
This looks OK to me, but I'd appreciate another review, especially on the subsystem tests.  Maybe there are too many?  @yevgeny-shnaidman @ori-amizur please take a look.

File: subsystem/cluster_test.go [hunk 2]
```diff
@@ -4010,3 +4010,346 @@ var _ = Describe("disk encryption", func() {
...
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	It("install cluster with oVirt hosts - only masters", func() {
+		By("set hosts with hw info and setup essential host steps")
+		inventoryOvirtInfo := getoVirtInventory()
+		ips := hostutil.GenerateIPv4Addresses(3, defaultCIDRv4)
+		h1 := registerNode(ctx, clusterID, "h1", ips[0])
```
Comment (yevgeny-shnaidman) [inline subsystem/cluster_test.go:4049]: registerNode already contains generateEssentialHostStepsWithInventory (default one). I think that you could probably use registerHost

**Comment by bkhizgiy** [comment]
@yevgeny-shnaidman all comments were addressed and the tests are passing again, can you please take a final look?

**Pull request merged**